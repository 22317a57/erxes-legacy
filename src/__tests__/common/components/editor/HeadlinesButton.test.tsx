import { mount, shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import HeadlinesButton from '../../../../modules/common/components/editor/HeadlinesButton';

describe('HeadlinesButton component', () => {
  const defaultProps = {
    onOverrideContent: e => null
  };

  test('renders HeadlinesButton successfully', () => {
    shallow(<HeadlinesButton {...defaultProps} />);
  });

  test('renders successfully with default value', () => {
    const wrapper = mount(<HeadlinesButton {...defaultProps} />);
    const props = wrapper.props();

    expect(props).toMatchObject(defaultProps);
  });

  test('snapshot matches', () => {
    const rendered = renderer
      .create(<HeadlinesButton {...defaultProps} />)
      .toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
