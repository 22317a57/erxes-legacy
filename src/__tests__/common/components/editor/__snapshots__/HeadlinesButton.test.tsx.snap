// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadlinesButton component snapshot matches 1`] = `
<button
  className="sc-bdVaJa ehclgY"
  onClick={[Function]}
>
  H
</button>
`;
